### PR TITLE
fix(14912): Fix helm valuesObject with ApplicationSet

### PR DESCRIPTION
**Pull request by speedfl**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

Fixes [ISSUE #14912]

This fix crash loop backoff when using helm valuesObject with ApplicationSet

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: applicationset/utils/utils.go [hunk 1]
```diff
@@ -99,10 +123,15 @@ func (r *Render) deeplyReplace(copy, original reflect.Value, replaceMap map[stri
 			// specific case time
 			if currentType == "time.Time" {
 				copy.Field(i).Set(original.Field(i))
-			} else if currentType == "Raw.k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1" {
+				// Raw.k8s.io/apimachinery/pkg/runtime
```
Comment (ishitasequeira) [inline applicationset/utils/utils.go:126]: The comment does not give any explanation. Do we need this?

**Review by ishitasequeira — APPROVED** [review]
Overall changes LGTM!! Just a minor comment.

File: applicationset/utils/utils.go (hunk 1 above)
Comment (speedfl) [reply applicationset/utils/utils.go:126]: I will delete it. It was a mistake

Comment (speedfl) [reply applicationset/utils/utils.go]: Allo good!

**Comment by crenshaw-dev** [comment]
May be a flake, but `TestNamespacedDisableManifestGeneration`. Running again, we'll see if it fails again.

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Comment by 34fathombelow** [comment]
@crenshaw-dev I think we should cherry-pick this into 2.8 WDYT?

**Pull request merged**